### ISSUE TITLE
Flutter Gallery app does not run on the new macOS hardware using iOS 14 simulator

### ISSUE THREAD

**Issue by a-siva** [flutter/flutter#65132]
The Flutter Gallery app fails to build/run on the iOS 14 simulator when running on the new macOS hardware
(uses xCode_12_beta)

```
asiva@sivas-mac gallery % /Users/asiva/flutter/bin/flutter doctor
Downloading android-arm-profile/darwin-x64 tools...                 0.8s
Downloading android-arm-release/darwin-x64 tools...                 0.4s
Downloading android-arm64-profile/darwin-x64 tools...               0.5s
Downloading android-arm64-release/darwin-x64 tools...               0.4s
Downloading android-x64-profile/darwin-x64 tools...                 0.4s
Downloading android-x64-release/darwin-x64 tools...                 0.3s
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, 1.22.0-10.0.pre.50, on macOS 11.0 20A5299w, locale en-US)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 12.0)
[!] Android Studio (version 4.0)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] Connected device (1 available)

! Doctor found issues in 1 category.

asiva@sivas-mac gallery % /Users/asiva/flutter/bin/flutter run   
Launching lib/main.dart on iPhone SE (2nd generation) in debug mode...
 
Running Xcode build...                                                  
Xcode build done.                                            5.7s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    oah error: unhandled 16KB-unaligned mmap addr=0x0 len=0x21f3bb4 offset=0x1000 prot=1 flags=2
    clang: error: unable to execute command: Trace/BPT trap: 5
    clang: error: linker command failed due to signal (use -v to see invocation)
    Command Ld failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of
    supported deployment target versions is 9.0 to 14.0.99. (in target 'url_launcher' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of
    supported deployment target versions is 9.0 to 14.0.99. (in target 'shared_preferences' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of
    supported deployment target versions is 9.0 to 14.0.99. (in target 'path_provider' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of
    supported deployment target versions is 9.0 to 14.0.99. (in target 'package_info' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of
    supported deployment target versions is 9.0 to 14.0.99. (in target 'Flutter' from project 'Pods')

Could not build the application for the simulator.
Error launching application on iPhone SE (2nd generation).
asiva@sivas-mac gallery % 
```

**Comment by TahaTesser** [comment]
Hi @a-siva 
Can you please provide `flutter run --verbose` in a `.txt`t file and attach it in the comment 
Thank you

Relate to https://github.com/flutter/flutter/issues/61047

**Comment by jmagman** [comment]
This is https://github.com/flutter/flutter/issues/64502, the App.framework and Flutter.framework needs an arm simulator slice..  @a-siva is the dart team ready to pass their DTK off to someone on the Flutter engine team?
\cc @zanderso

**Comment by jmagman** [comment]
@godofredoc did we ever get that DTK hooked up in the lab somewhere?

**Comment by godofredoc** [comment]
No, we didn't. Actually it is still on the google packages processing center, they didn't deliver it to the lab.

**Comment by a-siva** [comment]
[log.txt](https://github.com/flutter/flutter/files/5171740/log.txt)

**Comment by a-siva** [comment]
> Hi @a-siva
> Can you please provide `flutter run --verbose` in a `.txt` file and attach it in the comment
> Thank you
> 
> Relate to #61047

Attached log file

**Comment by pcsosinski** [comment]
adding this to the 1.22 milestone as we'll need it in that release when it is done.

**Comment by alexmarkley** [comment]
@jmagman @godofredoc Hey, any update on the porting process? I'm also a DTK program member and I want to help with the porting process -- anything from testing to contributing patches. (See also https://github.com/dart-lang/sdk/issues/43477 and https://github.com/dart-lang/sdk/issues/43476 )

Is there any (internal or external) coordination process for folks who are under Apple's DTK agreement and want to help?

Thanks!

**Comment by alexmarkley** [comment]
I don't want to open a duplicate issue, so I'll leave a comment here. I _think_ this is the same issue, but I'll leave that to the team to decide.

Following instructions here: https://flutter.dev/docs/get-started/install/macos#create-and-run-a-simple-flutter-app

Output of `flutter doctor -v`:
```
DTK:helloWorldOne alex$ flutter doctor -v
[✓] Flutter (Channel master, 1.22.0-10.0.pre.264, on macOS 11.0 20A5374g x86_64, locale en-US)
    • Flutter version 1.22.0-10.0.pre.264 at /Users/alex/Development/Flutter/flutter
    • Framework revision 3e6863ab2b (10 hours ago), 2020-09-18 21:05:27 -0700
    • Engine revision 2abe69c608
    • Dart version 2.10.0 (build 2.10.0-136.0.dev)

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Applications/Xcode-beta.app/Contents/Developer
    • Xcode 12.2, Build version 12B5018i
    • CocoaPods version 1.9.3

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] Connected device (1 available)
    • iPhone SE (2nd generation) (mobile) • 626585CA-3271-4B70-A974-271E361AC12E • ios • com.apple.CoreSimulator.SimRuntime.iOS-14-2 (simulator)

! Doctor found issues in 2 categories.
DTK:helloWorldOne alex$ 
```

(Note that I'm using the latest master from today.)

Results of `flutter run`:
```
DTK:helloWorldOne alex$ flutter run
Launching lib/main.dart on iPhone SE (2nd generation) in debug mode...
Running Xcode build...                                                  
 └─Compiling, linking and signing...                         0.9s
Xcode build done.                                            8.5s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    rosetta error: unhandled 16KB-unaligned mmap addr=0x0 len=0x21f5a70 offset=0x1000 prot=1 flags=2
    clang: error: unable to execute command: Trace/BPT trap: 5
    clang: error: linker command failed due to signal (use -v to see invocation)
    Command Ld failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone SE (2nd generation).
DTK:helloWorldOne alex$ 
```

**Comment by jmagman** [comment]
This issue is a duplicate of #64502, there is no Flutter framework arm simulator slice.
